bazaarvoice/emodb
# Add Postman tests  for all endpoints under /sor/1/_table/{table}

## Github Issue #315 

-

## What Are We Doing Here?
We are extending automation testing coverage by adding postman tests. This commit is covering all kind of checks for all endpoints under  /sor/1/_table/{table} api path. 

Feature coverage is additionally updated in our (for now internal) testing artefacts.


## How to Test and Verify

1. Check out this PR
2. Start emodb on local
3. Add api-key with "permissions":["*|read|__"]}' 

curl -s -XPOST 'http://localhost:8080/uac/1/role/reserved/postman_without_permissions' \
      -H "Content-Type: application/x.json-create-role" -H "X-BV-Api-Key:local_admin"\
      -d '{"name":"Postman without permissions","description":"Postman without permissions","permissions":["*|read|__"]}'

curl -s -XPOST "http://localhost:8080/uac/1/api-key" \
    -H "Content-Type: application/x.json-create-api-key" -H "X-BV-Api-Key:local_admin" \
    -d '{"owner":"emodb-dev@bazaarvoice.com","description":"Postman Without Permissions API key","roles":[{"group":"reserved","id":"postman_without_permissions"}]}' | jq .

4. Open postman 
5. Import "EmoDB Tests.postman_collection.json" as a collection
5. Import "sor-1-_table.postman_environment.json" as environment variables
6. Set 'api-key-no-rights' environment variable to value of the key that has been generated in step 3
7. Use runner for executing postman collection

Expected result: All tests are successfully executed on local environment

## Risk

`Low`
There are no changes to application code base

### Required Testing

All newly added postman tests should be executed without failures.

### Risk Summary

-

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
